src/structurizr/api/structurizr_client.py

```diff
@@ -99,7 +100,11 @@ def __repr__(self) -> str:
         )
 
     def __enter__(self):
-        """Enter a context by locking the corresponding remote workspace."""
+        """Enter a context by locking the corresponding remote workspace.
+
+        Note: this method is deprecated in favour of lock(), and will be removed in a
+        future relesae.
```

Comment:
Sphinx has [a nice directive](https://www.sphinx-doc.org/en/master/usage/restructuredtext/directives.html#directive-deprecated) for this sort of thing. I haven't found anything like it for MkDocs yet. Either way, I suspect that since it's a dunder method, hardly anyone will look at the docstring. Instead, I suggest to create a `DeprecationWarning` when this method is called.

```py
warnings.warn("Using the `StructurizrClient` in a context is deprecated since version <next release> and will be removed in a future release. Please use the preferred method `.lock()` instead.", DeprecationWarning)
```

Reply:
0.4.1 for next release?

Reply:
Let's make it 0.5